src/user.rs

```diff
@@ -122,6 +129,10 @@ impl User {
                     events.push(SyncEvent::Invite(invited_room));
                 }
 
+                for joined_room in &joined_rooms {
+                    events.push(SyncEvent::UnreadRoom(joined_room.clone()));
```

Comment:
This will make it read all messages on join, right? For a user just logged in would it make it read all rooms though?

Reply:
It will queue the rooms so next iterations before start adding friends or sending messages, will read from rooms one by one (not all at once).

But considering that several users will be in the same state (requesting messages from rooms), I guess it can work without overloading the endpoint and behave similar to kernel.